<launch>

    <arg name="prefix" default=""/>
    
    <param name="robot_description" 
        textfile="$(find centauro_urdf)/urdf/centauro.urdf"/>
    
    <param name="robot_description_semantic"
        textfile="$(find centauro_srdf)/srdf/centauro.srdf"/>
    
    <param name="cartesian/problem_description" 
        textfile="$(find ReGoVis)/configs/centauro_stack.yaml"/>
    
    <include file="$(find cartesian_interface)/launch/cartesio.launch">
        <arg name="rate" value="1000.0"/>
        <arg name="prefix" value="$(arg prefix)"/>
        <arg name="markers" value="false"/>
    </include>

    <node pkg="ReGoVis" type="image_processing_node.py" name="image_processing_node" output="screen">
    	<param name="depth" value="1.5"/>
	<remap from="camera_info" to="/D435i/D435i_camera/infra1/camera_info"/>
        <remap from="image_raw" to="/D435i/D435i_camera/color/image_raw"/>
        <remap from="image_processing/visual_features" to="cartesian/visual_servoing_D435i_camera_color_optical_frame/features"/>
        <remap from="desired_visual_features" to="reference_governor/desired_features"/>
        <remap from="image_processing/reference_features" to="cartesian/visual_servoing_D435i_camera_color_optical_frame/desired_features"/>
    </node> 

    <node pkg="ReGoVis" type="reference_governor.py" name="reference_governor" output="screen">
    	<remap from="/camera/rgb/camera_info" to="/D435i/D435i_camera/infra1/camera_info"/>
	    <remap from="reference_governor/reference_features" to="cartesian/visual_servoing_D435i_camera_color_optical_frame/desired_features"/>
    </node>

    <rosparam param="regovis_MPC_sampling_time">0.1</rosparam>
    <rosparam param="regovis_preview_window_size">10</rosparam> 
    <rosparam param="regovis_visual_servoing_link">"visual_servoing_D435i_camera_color_optical_frame"</rosparam>
</launch>
